src/modules/rust.rs

```diff
@@ -125,26 +125,40 @@ fn extract_toolchain_from_rustup_override_list(stdout: &str, cwd: &Path) -> Opti
 
 fn find_rust_toolchain_file(context: &Context) -> Option<String> {
     // Look for 'rust-toolchain' as rustup does.
-    // https://github.com/rust-lang/rustup.rs/blob/d84e6e50126bccd84649e42482fc35a11d019401/src/config.rs#L320-L358
-
-    fn read_first_line(path: &Path) -> Option<String> {
-        let content = fs::read_to_string(path).ok()?;
-        let line = content.lines().next()?;
-        Some(line.trim().to_owned())
+    // https://github.com/rust-lang/rustup/blob/89912c4cf51645b9c152ab7380fd07574fec43a3/src/config.rs#L546-L616
+
+    fn read_channel(path: &Path) -> Option<String> {
+        let contents = fs::read_to_string(path).ok()?;
+
+        match contents.lines().count() {
+            0 => None,
+            1 => Some(contents.trim().to_owned()),
```

Comment:
This still needs the `is_empty` check.
```suggestion
            1 => Some(contents.trim()).filter(|s| !s.is_empty()).map(|s| s.to_owned()),
```

Reply:
I think this would apply to the TOML version as well, as there could be just an empty string like `channel = ""`.
So I will apply the logic to both of them.